jazz_create-serverless-service/index.js

```diff
@@ -57,107 +56,207 @@ module.exports.handler = (event, context, cb) => {
-        }
 
-        // Add rate expression to the propertiesObject;
-        if (event.body.service_type === "function") {
-            if (event.body.rateExpression !== undefined) {
-                var cronExpValidator = CronParser.validateCronExpression(event.body.rateExpression);
+    function serviceOnboarding(event, config, service_id) {
+        return new Promise((resolve, reject) => {
+            try {
+                var base_auth_token = "Basic " + new Buffer(util.format("%s:%s", config.SVC_USER, config.SVC_PASWD)).toString("base64");
+                var userlist = "";
+                var approvers = event.body.approvers;
+                var domain = (event.body.domain || "").toLowerCase();
+                var service_name = event.body.service_name.toLowerCase();
 
-                // Validate cron expression. If valid add it to propertiesObject, else throw error
-                if (cronExpValidator.result === 'valid') {
-                    propertiesObject['rateExpression'] = event.body.rateExpression;
+                var bitbucketName = service_name;
+                if (domain.length) {
+                    bitbucketName = domain + "_" + bitbucketName;
+                }
+                userlist = approvers.reduce(function (stringSoFar, approver) {
+                    return stringSoFar + util.format("name=%s&", approver);
+                }, "");
 
-                    // enableEventSchedule is added here as an additional feature. It will be passed on to deployment-env.yml
-                    // If it is set as false it will be picked by serverless and event schedule will be disabled.
-                    // If the user chooses to stop the cron event, he can just disable and then re-enable it instead of deleting.
-                    if (event.body.enableEventSchedule === false) {
-                        propertiesObject['enableEventSchedule'] = event.body.enableEventSchedule;
+                var input = {
+                    token: config.BUILD_TOKEN,
+                    admin_group: userlist,
+                    service_id: service_id
+                };
+                request({
+                    url: config.JOB_BUILD_URL,
+                    method: 'POST',
+                    headers: {
+                        "Authorization": base_auth_token
+                    },
+                    qs: input
+                }, function (err, response, body) {
+                    if (err) {
+                        logger.error('Error while starting Jenkins job: ' + err);
+                        reject(err);
                     } else {
-                        // enable by default
-                        propertiesObject['enableEventSchedule'] = true;
+                        if (response.statusCode <= 299) { // handle all 2xx response codes as success
+                            var messageToBeSent = "Your service code will be available at " + config.BIT_BUCKET_URL + bitbucketName + "/browse";
```

Comment:
looks like this code will not work for gitlab. Please publish a new PR where it works for gitlab.

Reply:
On thinking about this more, I don't believe we have to send back to the caller info about their repo. If service on boarding handles creation of git repo and stuff, we should send that info to service handler which can update the catalog with the repo info.